### PR TITLE
hifiasm version bump to 0.18.5

### PR DESCRIPTION
**Pull request by abueg**
draft PR, new hifiasm update today -- biocondabot has merged the PR for it a couple hours earlier

### CODE REVIEW

**Comment by abueg** [comment]
i think the conda update is live!

**Comment by bgruening** [comment]
The new linter complains about `<has_size value="0"/>` value 0 is not allowed anymore. Do you recall why we test files with 0 size?

**Comment by abueg** [comment]
hmm no i didn't write that test, but looking at that test case it seems it would just be a case where everything is in the primary gfa so the alternate gfa is made but empty; should i edit the test to just check if the alternate gfa exists?

**Comment by bgruening** [comment]
Yes I think this would be nice.
@gallardoalba any other idea?

File: tools/hifiasm/hifiasm.xml [hunk 1]
```diff
@@ -318,11 +314,7 @@
             <output name="raw_unitigs" file="hifiasm-out2-raw.gfa" ftype="gfa1" />
             <output name="processed_unitigs" file="hifiasm-out2-processed.gfa" ftype="gfa1" />
             <output name="primary_contig_graph" file="hifiasm-out2-primary.gfa" ftype="gfa1" />
-            <output name="alternate_contig_graph" ftype="gfa1">
-                <assert_contents>
-                    <has_size value="0"/>
-                </assert_contents>
-            </output>
+            <output name="alternate_contig_graph" ftype="gfa1" />
```
Comment (gallardoalba) [inline tools/hifiasm/hifiasm.xml:317]:
```suggestion
```

File: tools/hifiasm/hifiasm.xml [hunk 2]
```diff
@@ -305,11 +305,7 @@
             <output name="raw_unitigs" file="hifiasm-out1-raw.gfa" ftype="gfa1" />
             <output name="processed_unitigs" file="hifiasm-out1-processed.gfa" ftype="gfa1" />
             <output name="primary_contig_graph" file="hifiasm-out1-primary.gfa" ftype="gfa1" />
-            <output name="alternate_contig_graph" ftype="gfa1">
-                <assert_contents>
-                    <has_size value="0"/>
-                </assert_contents>
-            </output>
+            <output name="alternate_contig_graph" ftype="gfa1" />
```
Comment (gallardoalba) [inline tools/hifiasm/hifiasm.xml:308]:
```suggestion
```

File: tools/hifiasm/hifiasm.xml [hunk 3]
```diff
@@ -333,11 +325,7 @@
...
```
Comment (gallardoalba) [inline tools/hifiasm/hifiasm.xml:328]:
```suggestion
```

File: tools/hifiasm/hifiasm.xml [hunk 4]
```diff
@@ -390,26 +378,10 @@
...
-            </output>
-            <output name="hap2_contigs">
-                <assert_contents>
-                    <has_size value="0"/>
-                </assert_contents>
-            </output>
+            <output name="raw_unitigs_trio" ftype="gfa1" />
+            <output name="processed_unitigs_trio" ftype="gfa1" />
+            <output name="hap1_contigs" />
+            <output name="hap2_contigs" />
```
Comment (gallardoalba) [inline tools/hifiasm/hifiasm.xml:384]:
```suggestion
            <assert_command>
                <has_text text="-1 hap1.yak"/>
                <has_text text="-2 hap2.yak"/>
                <has_text text="--primary"/>
            </assert_command>
```

File: tools/hifiasm/hifiasm.xml [hunk 5]
```diff
@@ -423,11 +395,7 @@
             <output name="raw_unitigs" file="hifiasm-out3-raw.gfa" ftype="gfa1" />
             <output name="processed_unitigs" file="hifiasm-out3-processed.gfa" ftype="gfa1" />
             <output name="primary_contig_graph" file="hifiasm-out3-primary.gfa" ftype="gfa1" />
-            <output name="alternate_contig_graph" ftype="gfa1">
-                <assert_contents>
-                    <has_size value="0"/>
-                </assert_contents>
-            </output>
+            <output name="alternate_contig_graph" ftype="gfa1" />
```
Comment (gallardoalba) [inline tools/hifiasm/hifiasm.xml:398]:
```suggestion
```

File: tools/hifiasm/hifiasm.xml [hunk 6]
```diff
@@ -441,11 +409,7 @@
             <output name="raw_unitigs" file="hifiasm-out4-raw.gfa" ftype="gfa1" />
             <output name="processed_unitigs" file="hifiasm-out4-processed.gfa" ftype="gfa1" />
             <output name="primary_contig_graph" file="hifiasm-out4-primary.gfa" ftype="gfa1" />
-            <output name="alternate_contig_graph" ftype="gfa1">
-                <assert_contents>
-                    <has_size value="0"/>
-                </assert_contents>
-            </output>
+            <output name="alternate_contig_graph" ftype="gfa1" />
```
Comment (gallardoalba) [inline tools/hifiasm/hifiasm.xml:412]:
```suggestion
```

File: tools/hifiasm/hifiasm.xml [hunk 7]
```diff
@@ -459,11 +423,7 @@
             <output name="raw_unitigs" file="hifiasm-out5-raw.gfa" ftype="gfa1" />
             <output name="processed_unitigs" file="hifiasm-out5-processed.gfa" ftype="gfa1" />
             <output name="primary_contig_graph" file="hifiasm-out5-primary.gfa" ftype="gfa1" />
-            <output name="alternate_contig_graph"  ftype="gfa1">
-                <assert_contents>
-                    <has_size value="0"/>
-                </assert_contents>
-            </output>
+            <output name="alternate_contig_graph"  ftype="gfa1" />
```
Comment (gallardoalba) [inline tools/hifiasm/hifiasm.xml:426]:
```suggestion
```

File: tools/hifiasm/hifiasm.xml [hunk 8]
```diff
@@ -477,11 +437,7 @@
             <output name="raw_unitigs" file="hifiasm-out6-raw.gfa" ftype="gfa1" />
             <output name="processed_unitigs" file="hifiasm-out6-processed.gfa" ftype="gfa1" />
             <output name="primary_contig_graph" file="hifiasm-out6-primary.gfa" ftype="gfa1" />
-            <output name="alternate_contig_graph"  ftype="gfa1">
-                <assert_contents>
-                    <has_size value="0"/>
-                </assert_contents>
-            </output>
+            <output name="alternate_contig_graph"  ftype="gfa1" />
```
Comment (gallardoalba) [inline tools/hifiasm/hifiasm.xml:440]:
```suggestion
```

File: tools/hifiasm/hifiasm.xml [hunk 9]
```diff
@@ -495,11 +451,7 @@
             <output name="raw_unitigs" file="hifiasm-out7-raw.gfa" ftype="gfa1" />
             <output name="processed_unitigs" file="hifiasm-out7-processed.gfa" ftype="gfa1" />
             <output name="primary_contig_graph" file="hifiasm-out7-primary.gfa" ftype="gfa1" />
-            <output name="alternate_contig_graph"  ftype="gfa1">
-                <assert_contents>
-                    <has_size value="0"/>
-                </assert_contents>
-            </output>
+            <output name="alternate_contig_graph"  ftype="gfa1" />
```
Comment (gallardoalba) [inline tools/hifiasm/hifiasm.xml:454]:
```suggestion
```

File: tools/hifiasm/hifiasm.xml [hunk 10]
```diff
@@ -513,11 +465,7 @@
             <output name="raw_unitigs" file="hifiasm-out8-raw.gfa" ftype="gfa1" />
             <output name="processed_unitigs" file="hifiasm-out8-processed.gfa" ftype="gfa1" />
             <output name="primary_contig_graph" file="hifiasm-out8-primary.gfa" ftype="gfa1" />
-            <output name="alternate_contig_graph" ftype="gfa1">
-                <assert_contents>
-                    <has_size value="0"/>
-                </assert_contents>
-            </output>
+            <output name="alternate_contig_graph" ftype="gfa1" />
```
Comment (gallardoalba) [inline tools/hifiasm/hifiasm.xml:468]:
```suggestion
```

**Comment by bgruening** [comment]
Do you know the `planemo test --update_test_data` trick?

**Comment by abueg** [comment]
.... now i do 😅

**Review by bgruening — APPROVED** [review]

**Comment by abueg** [comment]
image
works fine locally -- looking at the tool tests output artifact on here, it seems test 3 ran with 2 threads but my test was looking for "-t 1" flag in the log -- let me edit it to search for the command text that doesn't include the thread parameter in case that can change

**Review by bgruening — APPROVED** [review]

**Comment by bgruening** [comment]
Thanks a bunch! Automatic update will deploy it this weekend :)

**Pull request merged**